addons/website_event_track_quiz/controllers/main.py

```diff
@@ -0,0 +1,144 @@
+        }
+
+    def _get_event_track_quiz_data(self, track):
+        track_visitor = self._get_event_track_visitor(track)
+        values = {
+            'slide_questions': [{
+                'id': question.id,
+                'question': question.question,
+                'answer_ids': [{
+                    'id': answer.id,
+                    'text_value': answer.text_value,
+                    'is_correct': answer.is_correct if track_visitor.quiz_completed or request.website.is_publisher() else None,
+                    'comment': answer.comment if request.website.is_publisher else None
+                } for answer in question.sudo().answer_ids],
+            } for question in track.quiz_id.question_ids]
+        }
+        if 'slide_answer_quiz' in request.session:
+            slide_answer_quiz = json.loads(request.session['track_answer_quiz'])
+            if str(track.id) in slide_answer_quiz:
+                values['session_answers'] = slide_answer_quiz[str(track.id)]
+        values.update(self._get_event_track_visitor(track))
+        return values
+
+    def _get_event_track_visitor(self, track):
+        partner = request.env.user.partner_id
+        event_track_visitor = request.env['event.track.visitor'].sudo().search([('track_id', '=', track.id), ('partner_id', '=', partner.id)])
+        if not event_track_visitor:
+            values = {
+                'partner_id': partner.id,
+                'quiz_completed': False,
+                'points_gained': 0,
+                'track_id': track.id
+            }
+            event_track_visitor = request.env['event.track.visitor'].sudo().create(values)
+        return event_track_visitor
+
+    def _get_quiz_answers_details(self, track, answer_ids):
+        all_questions = request.env['event.quiz.question'].sudo().search([('quiz_id', '=', track.quiz_id.id)])
+        user_answers = request.env['event.quiz.question.answer'].sudo().search([('id', 'in', answer_ids)])
+
+        if user_answers.mapped('question_id') != all_questions:
+            return {'error': 'quiz_incomplete'}
+
+        user_bad_answers = user_answers.filtered(lambda answer: not answer.is_correct)
+        user_good_answers = user_answers - user_bad_answers
+        return {
+            'user_bad_answers': user_bad_answers,
+            'user_good_answers': user_good_answers,
+            'user_answers': user_answers,
+            'points': sum([answer.awarded_points for answer in user_good_answers])
+        }
+
+    @http.route('/event/track/quiz/get', type="json", auth="public", website=True)
+    def event_track_quiz_get(self, track_id):
+        fetch_res = self._fetch_track(track_id)
+        if fetch_res.get('error'):
+            return fetch_res
+        track = fetch_res['track']
+        return self._get_event_track_quiz_data(track)
+
+    @http.route('/event/track/quiz/submit', type="json", auth="public", website=True)
```

Comment:
AlRouting: event_track/quiz/submit (utility json -> simplified)

I would also give event_id (to always be under an event: event_id, track_id -> check event is accessible and matches track event)